File: test/utils_test.go
package test

import (
	"coura/klippify/internal/utils"
	"testing"
)

func TestMatchOrUnknown(t *testing.T) {

	tests := []struct {
		name     string
		a, b     string
		expected string
	}{
		{"Should extract the string", "Hello, World!", "World", "World"},
		{"Should return 'UNKNOWN' when the string is not found", "Hello, World!", "Dog", utils.UNKNOWN},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := utils.MatchOrUnknown(tt.a, tt.b)

			if result != tt.expected {
				t.Errorf("Expected %s, got %s", tt.expected, result)
			}
		})
	}
}

File: src/utils/utils.test.ts
import { UNKNOWN } from "../constants";
import { matchOrUnkown } from "./utils";

describe("Unit tests for utils", () => {
    it("Should match the pattern", () => {
        const text = "Hello World";
        const pattern = /Hello/;

        const result = matchOrUnkown(text, pattern);
        expect(result).toBe("Hello");
    });

    it("Should return UNKNOWN when the pattern does not match", () => {
        const text = "Hello World";
        const pattern = /Goodbye/;

        const result = matchOrUnkown(text, pattern);
        expect(result).toBe(UNKNOWN);
    });
});